### ISSUE TITLE
"System - HTTP Headers" plugin results in error with google maps embedded on page after clicking on url in way point

### ISSUE THREAD

**Issue by Hans-63** [joomla/joomla-cms#36682]
On my website I have an embedded map of Google Maps with various way-points (houses for sale - https://www.htm63.com/in-english/our-properties-on-the-map.html). In the description of such a way-point there are url's which refer to a page on my site with the description of the house in several languages.
This worked fine in Joomla 3 but in Joomla 4 an error message appeared after clicking on a url: "400. That's an error.
The server cannot process the request because it is malformed. It should not be retried. That's all we know."

### Steps to reproduce the issue

Emdded google maps map with way-point containing url and 'System - HTTP headers' plugin turned on then click on url... : error message

### Expected result

New page is page reffered bij url and not the error page...

### Actual result

Error 400 page

### System information (as much as possible)

This problem can be solved by turning off the "System - HTTP Headers plugin". Clicking on the url will then bring you to the desired page....

Because I would like to have the corresponding security, but also want to continue to use google maps, it would be nice if this would work as in Joomla 3..... So "System - HTTP Headers" plugin" turned 'on' and a working url-click with the embedded google maps.

### Additional comments

PHP Built On	Linux dh29280.eatserver.nl 3.10.0-962.3.2.lve1.5.61.el7.x86_64 #1 SMP Wed Aug 18 07:50:15 UTC 2021 x86_64
Database Type	mysql
Database Version	10.5.13-MariaDB
Database Collation	utf8_general_ci
Database Connection Collation	utf8mb4_general_ci
Database Connection Encryption	None
Database Server Supports Connection Encryption	No
PHP Version	7.4.27
Web Server	Apache
WebServer to PHP Interface	cgi-fcgi
Joomla! Version	Joomla! 4.0.5 Stable [ Furaha ] 11-December-2021 20:52 GMT

**Comment by wilsonge** [comment]
@zero-24 this one's for you i think

**Comment by zero-24** [comment]
hmm that sounds interesting. Can you reproduce the issue on J3 when using the 3.x version of the plugin: https://github.com/zero-24/plg_system_httpheader/releases/latest

And can you please post the exact settings set in the plugin and try to enable and disable them step by step?
Does it also happen when you use the direct URL and not an bit.ly URL?
Does the URL behind the bit.ly URL shortener use HTTPS and not HTTP?

**Comment by Hans-63** [comment]
Thank you for your support and for giving me directions on how to solve/find the problem. It seems to be solved:

The old J3 site was no longer available but another site of mine was. Here I installed the same page with google maps embedded and it worked fine......
However, here the plugin was not installed... I've installed it and all works as intended.... (See attachment for screenshots of settings)
Next I adjusted the settings in Joomla 4 of the plugin as much as possible to the J3 settings and then I found out that the 'Same-origin' option at 'Cross-Origin-Opener-Policy' caused the problems and that the other options did give the intended result.
I had also tried changing the URL used before without good results (bitly / http / https).

Thanks a lot and Ihope this helps others!!


[joomla https-headers.pdf](https://github.com/joomla/joomla-cms/files/7876851/joomla.https-headers.pdf)
